### PR TITLE
Bump to dep v0.4.1

### PR DESCRIPTION
**Pull request by freeformz**
Make dep versions "configurable" (at the buildpack level for now).

### CODE REVIEW

File: files.json [hunk 1]
```diff
@@ -275,6 +275,17 @@
     "dep-linux-amd64": {
         "URL": "https://github.com/golang/dep/releases/download/v0.3.1/dep-linux-amd64",
         "SHA": "408b970c0acacad309d350fc46d718e4a8885b588eca71f4d73e5d6c521ef944",
+        "LocalName": "dep",
+        "Comment": "This is here for backwards compat. Remove after a while."
+    },
+    "dep-v0.3.1-linux-amd64": {
+        "URL": "https://github.com/golang/dep/releases/download/v0.3.1/dep-linux-amd64",
+        "SHA": "408b970c0acacad309d350fc46d718e4a8885b588eca71f4d73e5d6c521ef944",
```
Comment (Xe) [inline files.json:283]:
```console
[~] : wget -O- https://github.com/golang/dep/releases/download/v0.3.1/dep-linux-amd64 | sha256sum                                                                                                                                     00:26:29
HTTP request sent, awaiting response... 200 OK
Length: 7661152 (7.3M) [application/octet-stream]
Saving to: ‘STDOUT’

-                                                           100%[=========================================================================================================================================>]   7.31M  8.53MB/s    in 0.9s
                                                                                                                                                                                                                                              2018-01-25 00:26:43 (8.53 MB/s) - written to stdout [7661152/7661152]

408b970c0acacad309d350fc46d718e4a8885b588eca71f4d73e5d6c521ef944  -
```

File: files.json [hunk 2]
```diff
@@ -275,6 +275,17 @@
...
+        "LocalName": "dep"
+    },
+    "dep-v0.4.0-linux-amd64": {
+        "URL":"https://github.com/golang/dep/releases/download/v0.4.0/dep-linux-amd64",
+        "SHA":"228088ab0fe15dff6779f4e48223a6dd39544d3348b2f43b28b96eae27f341b5",
```
Comment (Xe) [inline files.json:288]:
```console
[~] : wget -O- https://github.com/golang/dep/releases/download/v0.4.0/dep-linux-amd64 | sha256sum
HTTP request sent, awaiting response... 200 OK
Length: 8486624 (8.1M) [application/octet-stream]
Saving to: ‘STDOUT’

-                                                           100%[=================================

2018-01-25 00:27:54 (6.23 MB/s) - written to stdout [8486624/8486624]

228088ab0fe15dff6779f4e48223a6dd39544d3348b2f43b28b96eae27f341b5  -
```

**Review by Xe — APPROVED** [review]
Manually verified the shas and they match (see review feedback).

**Comment by danp** [comment]
Heads up that dep 0.4.1 has already been released.

**Pull request merged**